### PR TITLE
Front end

### PR DESCRIPTION
**Pull request by Fsher07**
- Create a new React app called 'hello-react-front-end'
- Initialize your project with Git.
- Create the `App` component with react-router.
- Create the `Greeting` component that will display a greeting. Set it up as a route in your App component.
- Create a store, an action and a reducer that will connect to you API endpoint to get the random greeting.
- Display the random greeting in your `Greeting` component.

[Rails API](https://github.com/Fsher07/rails-back-end-api)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,56 +1,18 @@
-import React from 'react';
-import logo from './logo.svg';
-import { Counter } from './features/counter/Counter';
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchGreetings } from './features/greeting/greetingSlice';
 import './App.css';
 
 function App() {
```
Comment (FrederickMih) [inline src/App.js:6]: - [ ] Based on the requirements of this project, the App component should have routes(from react-router) that lead to the Greeting component hence you'll require the `Greeting` component. Let's kindly fix this.

File: src/features/greeting/greetingSlice.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+);
+
+export const counterSlice = createSlice({
+  name: 'greeting',
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    builder
+      .addCase(fetchGreetings.pending, (state) => {
+        // eslint-disable-next-line no-param-reassign
```
Comment (FrederickMih) [inline src/features/greeting/greetingSlice.js]: - [ ] Let's kindly fix these Linters issues. Don't disable them. Linters are great tools that flag programming errors. Kindly fix all in this file.

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @Fsher07 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️

- PR has a title and summary :heavy_check_mark: 
- GitHub workflow has been followed :heavy_check_mark: 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/app/Greeting.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  return (
+    <div className="App">
+      <h1>
+        {!message.loading && message.message}
+      </h1>
+    </div>
+  );
+}
+
+export default Greeting;
```
Comment (ahmetbozaci) [inline src/app/Greeting.js:21]: - [ ] I cant see any message on the page. Kindly fix it.
image
NOTE: I run the rails server and ı see the JSON file. The rails side is working.
image

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# ♻️♻️♻️ STATUS: Changes Required  ♻️♻️♻️ 


@Fsher07  ✋🏼 
#### Great work on making the changes requested by a previous reviewer 👏 
#### You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
___ 

### 💯  Highlights 💯
- Your git-flow is correct. ✔️
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
___

### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
almost


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._
___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Pull request closed without merging**